typesafety/test_typing_type.yml

```diff
@@ -0,0 +1,26 @@
+- case: typing_type_correct
+  disable_cache: true
+  main: |
+    from typing import Type, TypeVar
+    from classes import typeclass
+
+    class _MyClass(object):
+        ...
+
+    class _MyClassType(Type[_MyClass]):
+        ...
+
+
+    @typeclass
+    def class_type(typ) -> Type:
+        ...
+
+    @class_type.instance(delegate=_MyClassType)
+    def _my_class_type(typ: Type[_MyClass]) -> Type:
+        return typ
+
+    class_type(_MyClass)
+    class_type(int)
+  out: |
+    main:7: error: Invalid base class "Type"
```

Comment:
@sobolevn I just need your help to figure out this or if it's ok to leave it as is!

Reply:
See https://github.com/python/mypy/issues/10962